### PR TITLE
Create an HttpsProxyAgent for plugin list if necessary

### LINKED ISSUE #5480: serverless plugin behind proxy
How can I use the command `serverless plugin list` behind proxy?  
I am getting this error:

```
request to https://raw.githubusercontent.com/serverless/plugins/master/plugins.json failed, reason: connect ECONNREFUSED 151.101.84.133:443
```

The command `serverless deploy` works fine with my proxy, so the proxy and/or proxy configuration should not be the problem.

**Versions**
```shell
$ node --version
v11.1.0

$ npm --version
6.4.1
```

**Environment variables**
```shell
$ env | grep -i proxy
NO_PROXY=localhost,127.0.0.0/8,::1
http_proxy=http://localhost:3128/
npm_config_https_proxy=http://localhost:3128/
https_proxy=http://localhost:3128/
no_proxy=localhost,127.0.0.0/8,::1
npm_config_proxy=http://localhost:3128/
```

**cURL**

Even if I `curl` it, it works fine with my proxy settings. Truncated for brevity.
```shell
$ curl --verbose https://raw.githubusercontent.com/serverless/plugins/master/plugins.json
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 3128 (#0)
* Establish HTTP proxy tunnel to raw.githubusercontent.com:443
> CONNECT raw.githubusercontent.com:443 HTTP/1.1
> Host: raw.githubusercontent.com:443
> User-Agent: curl/7.47.0
> Proxy-Connection: Keep-Alive
> 
< HTTP/1.1 200 Connection established
< Connection: close
< 
* Proxy replied OK to CONNECT request
* found 155 certificates in /etc/ssl/certs/ca-certificates.crt
* found 625 certificates in /etc/ssl/certs
* ALPN, offering http/1.1
* SSL connection using TLS1.2 / ECDHE_RSA_AES_128_GCM_SHA256
* 	 server certificate verification OK
* 	 server certificate status verification SKIPPED
* 	 common name: www.github.com (matched)
* 	 server certificate expiration date OK
* 	 server certificate activation date OK
* 	 certificate public key: RSA
* 	 certificate version: #3
* 	 subject: C=US,ST=California,L=San Francisco,O=GitHub\, Inc.,CN=www.github.com
* 	 start date: Thu, 23 Mar 2017 00:00:00 GMT
* 	 expire date: Wed, 13 May 2020 12:00:00 GMT
* 	 issuer: C=US,O=DigiCert Inc,OU=www.digicert.com,CN=DigiCert SHA2 High Assurance Server CA
* 	 compression: NULL
* ALPN, server accepted to use http/1.1
> GET /serverless/plugins/master/plugins.json HTTP/1.1
> Host: raw.githubusercontent.com
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
< Strict-Transport-Security: max-age=31536000
< X-Content-Type-Options: nosniff
< X-Frame-Options: deny
< X-XSS-Protection: 1; mode=block
< ETag: "53374f2ed929f6d42faf13ebb025e0ff985941f1"
< Content-Type: text/plain; charset=utf-8
< Cache-Control: max-age=300
< X-Geo-Block-List:
< X-GitHub-Request-Id: 447E:6762:1CABB8:1E679B:5BEAD6AF
< Content-Length: 45355
< Accept-Ranges: bytes
< Date: Tue, 13 Nov 2018 13:50:39 GMT
< Via: 1.1 varnish
< Connection: keep-alive
< X-Served-By: cache-bma1635-BMA
< X-Cache: MISS
< X-Cache-Hits: 0
< X-Timer: S1542117040.576321,VS0,VE107
< Vary: Authorization,Accept-Encoding
< Access-Control-Allow-Origin: *
< X-Fastly-Request-ID: 929450c7326d8ce7c7ffa13174cbd80c86ecc255
< Expires: Tue, 13 Nov 2018 13:55:39 GMT
< Source-Age: 0
< 
```

I asked it on stackoverflow also:
https://stackoverflow.com/questions/53282581/serverless-plugin-behind-proxy

### PR DESCRIPTION
**Pull request by dschep**
## What did you implement:

Closes #5480

## How did you implement it:

Create an `HttpsProxyAgent` to pass to `fetch` if any relevant env vars are set

## How can we verify it:

run `sls plugins list` with an `https_proxy` env var set.

## Todos:

- [ ] Write tests
- [ ] Write documentation
- [ ] Fix linting errors
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

**Comment by dschep** [comment]
Doh. sorry @lazize, it's `#` not `@`. So: `npm i -g serverless/serverless#sls-5480`

File: lib/plugins/plugin/lib/utils.js [hunk 1]
```diff
@@ -49,7 +50,20 @@ module.exports = {
...
+    // Use HTTPS Proxy (Optional)
+    const proxy = process.env.proxy
+      || process.env.HTTP_PROXY
+      || process.env.http_proxy
+      || process.env.HTTPS_PROXY
+      || process.env.https_proxy;
+
+    const options = {};
+    if (proxy) {
+      options.agent = new HttpsProxyAgent(url.parse(proxy));
```
Comment (lazize) [inline lib/plugins/plugin/lib/utils.js]: See the error below:

```
$ serverless plugin  list
 
  Reference Error ----------------------------------------
 
  url is not defined
 
     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information -----------------------------
     OS:                     linux
     Node Version:           11.1.0
     Serverless Version:     1.32.0
```

Comment (dschep) [reply lib/plugins/plugin/lib/utils.js]: Doh. Fixed. re-install and try again please.

**Comment by dschep** [comment]
Yeah. didn't test since you'd have to do a bunch of mocking sinc eit's explicitly network stuff.

**Pull request merged**